lib/puppet/node/environment.rb

```diff
@@ -82,8 +82,12 @@ def self.remote(name)
   # @param name [Symbol] The environment name
   def initialize(name, modulepath, manifest, config_version)
     @name = name
-    @modulepath = modulepath
-    @manifest = manifest
+    @modulepath = modulepath.map { |p| Puppet::FileSystem.expanded_path_string(p) }
+    if manifest && !manifest.is_a?(Symbol) && Puppet::FileSystem.exist?(manifest)
```

Comment:
What is up with the is_a?(Symbol) case? Is that for the NoManifest case ? If so I think it is best if it is explicit.